### ISSUE TITLE
Question about decomposition retalted codes

### ISSUE THREAD

**Issue by ccxi1008** [WuJH2001/swift4d#4]
Hi! I have question since the code seems inconsistent with the idea in the article.
The function` scene_reconstruction` end with           
```
 if iteration < final_iter:
                gaussians.optimizer.step()
                gaussians.optimizer.zero_grad(set_to_none = True)
                gaussians.hash_optimizer.step()
                gaussians.hash_optimizer.zero_grad(set_to_none = True)
```

which means all the gaussians no matter dynamic or static have hash deformation network and will be updated. 
When training, there are 3 steps in order:

```
    scene_reconstruction(dataset, opt, hyper, pipe, testing_iterations, saving_iterations,
                              checkpoint_iterations, checkpoint, debug_from,
                              gaussians, scene, "coarse", tb_writer, opt.coarse_iterations,timer,args)
    if opt.train_dynamic:
        train_dynamic ( pipe , opt,  scene,  gaussians , tb_writer ,args)
    
    scene_reconstruction(dataset, opt, hyper, pipe, testing_iterations, saving_iterations,
                         checkpoint_iterations, checkpoint, debug_from,
                         gaussians, scene, "fine", tb_writer, opt.iterations,timer, args)
```

The first time` scene_reconstruction` already mix all gaussians and all hash deformtion related things update . Then function `train_dynamic ` only update the dynamic labels. The second time `scene_reconstruction` again train all mixed gaussians with hash related things(deformation). How this matchthe idea that only dynamic GS use deformation network and static ones are original static GS? The codes seem to only identify the label but update all other parameters together with no difference for two types of GS. I wonder how can this match with "Dynamic
points undergo deformation using a spatio-temporal structure, as discussed in Sec.3.3. Finally, the deformed dynamic points are mixed with static points for rendering"
Thanks!

**Comment by WuJH2001** [comment]
You may want to take a look at lines 84–99 in `gaussian_renderer/__init__.py`. In the coarse stage, the deformation field is actually not optimized; it's only optimized together with the dynamic points during the fine stage.

**Comment by ccxi1008** [comment]
> You may want to take a look at lines 84–99 in `gaussian_renderer/__init__.py`. In the coarse stage, the deformation field is actually not optimized; it's only optimized together with the dynamic points during the fine stage.
> 
> > You can also add a condition before
> > `gaussians.hash_optimizer.step() gaussians.hash_optimizer.zero_grad(set_to_none=True)`
> > such as `if stage == "fine": gaussians.hash_optimizer.step() gaussians.hash_optimizer.zero_grad(set_to_none=True)`

I see！Thanks a lot!
